### ISSUE TITLE
steamtricks_fix_apply() manifest parse leaks values

### ISSUE THREAD

**Issue by boombatower** [steamtricks/steamtricks#15]
I was aware of this when committing, but there is no easy workaround.

The offending code in `steamtricks_fix_apply()` is:
```bash
eval $(keyvalue_parse "$manifest")
```

Since the variables are not prefixed with local (a possible solution) they are loaded into global variable space. I had attempt something more like what is in `steam_library_directories()`:
```bash
(
    eval $(keyvalue_parse "$STEAM_DIR/steamapps/libraryfolders.vdf")
    for var in $(compgen -A variable | grep -E "LibraryFolders_[0-9]+$") ; do
      eval echo \$$var/steamapps
    done
  )
```

That solution is cumbersome since fix_apply needs to access the data immediately. Something like:
```bash
read a b <<< $(
  eval ...
  echo
)
```

works, but run into space issue since games like "Portal 2" cause some issues in read. Using `$IFS` didn't seem to do the trick, likely because $() condenses to space separated. Alternatively, `keyvalue_parse()` could be modified to prefix lines with local as an option and additional prefix all variable names. The latter being useful when implementing a write function which could accept the variable prefix and write all variables with that prefix similarly to `C_` used in writing config variables.

For now this odd enough to solve that it's not worth the effort.

**Comment by boombatower** [comment]
A thought I had while implementing was using keyed arrays, but bash provides no mechanism for passing them between functions. As such the only real solution is to use a global which seems about as ugly, but is probably a cleaner approach and would get rid of the need for eval.

**Comment by boombatower** [comment]
An associative array makes the LibraryFolders grep a bit less elegant and ugly. Alternatively the function could be changed to append `local` before each definition, but would obviously only work when called from a function. This also makes the write() method potentially uglier. I'll play around with it.

**Comment by boombatower** [comment]
Solved by 9d0cb50, but could revisit the array when writing or down the road. More important not to hold onto extra memory than a fancier fix.
